src/main/java/com/epam/pages/popup/AcademicYearsAndVacationPopup.java

```diff
@@ -1,6 +1,112 @@
 package com.epam.pages.popup;
 
-import com.epam.pages.common.CommonPopup;
+import com.epam.helpers.ErrorMessagesProvider;
+import com.epam.helpers.SharedTestData;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.Select;
 
-public class AcademicYearsAndVacationPopup extends CommonPopup {
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+public class AcademicYearsAndVacationPopup extends CreatePopup {
+    @FindBy(id = "startDate")
+    private WebElement startDate;
+    @FindBy(id = "endDate")
+    private WebElement endDate;
+    @FindBy(className = "ui-datepicker-month")
+    private WebElement monthToSelect;
+    @FindBy(className = "ui-datepicker-year")
+    private WebElement yearToSelect;
+    @FindBy(xpath = "//select[@class='ui-datepicker-year']/option")
+    private List<WebElement> listOfYears;
+    @FindBy(xpath = "//td[contains(@class, 'disabled') and not(contains(@class, 'other-month'))]")
+    private List<WebElement> disabledDates;
+    @FindBy(xpath = "//div[@class='error']/p")
+    private WebElement errorMessageForWrongSelectedDates;
+
+    public boolean checkUIOfCreatePopupYearsSection() {
+        logger.info("Check UI of create popup in years section");
+        return uiHelper.checkElementsAreDisplayed(
+                title,
+                startDate,
+                endDate,
+                xButton,
+                saveButton
+        );
+    }
+
+    public boolean areDatesSelected() {
+        logger.info("Check if start date and end date are selected");
+        return uiHelper.areElementsSelected(
+                startDate,
+                endDate
+        );
+    }
+
+    public boolean checkErrorMessageForWrongSelectedDates() {
+        logger.info("Check if dates are selected wrong -> (start date < end date).");
+        return errorMessageForWrongSelectedDates.getText()
+                .equals(ErrorMessagesProvider.getWrongSelectedDatesErrMessage());
+    }
+
+    public boolean checkErrorMessageForForFillingLessThan30Days() {
+        logger.info("Check if dates are selected wrong -> (end date - start date < 30).");
+        return errorMessageForWrongSelectedDates.getText()
+                .equals(ErrorMessagesProvider.getLessThan30DaysErrMessage());
+    }
+
+    public void selectDate(int day, String month, int year, String field) {
+        logger.info("Fill {} date - {}/{}/{}", field, day, month, year);
+        if (field.contains("start")) {
+            uiHelper.clickOnWebElement(startDate);
+        } else if (field.contains("end")) {
+            uiHelper.clickOnWebElement(endDate);
+        }
```

Comment:
I think You can add else block too, because if the filed value won't be start or end, it won't be right.

Reply:
done